bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/service/ReadyService.java

```diff
@@ -0,0 +1,103 @@
+/**
+ * Copyright (c) 2014-2017 by the respective copyright holders.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.eclipse.smarthome.core.service;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+
+/**
+ * Registry for {@link ReadyMarker}s.
+ *
+ * Services may use the {@link ReadyService} in order to denote they have completed loading/processing something.
+ * <p>
+ * Interested parties may register as a tracker for {@link ReadyMarker}s. Optionally they can provide a
+ * {@link ReadyMarkerFilter} in order to restrict the {@link ReadyMarker}s they get notified for.
+ * <p>
+ * Alternatively, {@link #isReady(ReadyMarker)} can be used to check for any given {@link ReadyMarker}.
+ *
+ * @author Simon Kaufmann - initial contribution and API.
+ *
+ */
+@NonNullByDefault
```

Comment:
Did we already agree on using that annotation?

Reply:
No idea, I lost the overview...

Reply:
Check [this](https://github.com/eclipse/smarthome/pull/4080#issuecomment-328115241) to get an overview again :-)